### DISCUSSION TITLE
Add way to define a long string workflow_dispatch input

### DISCUSSION THREAD

**Discussion by gabimoncha** [community/community#12882] [Actions]
I'm trying to add new lines to the string input but the input box is unusable.

It would be nice to also add the option of `input` or `textarea` where the user can increase the height of the box and pressing `Enter` would add a new line in the field

**Comment by ethomson** [comment]
This is good feedback @gabrielmoncea - I'll talk with the team about it and make sure that it is on the backlog.

**Reply by Angelk90** [reply]
@ethomson : Is there any news?

**Reply by bahanurenis-tomtom** [reply]
I am also experiencing same problem is there any update yet?

**Reply by Timmmm** [reply]
Another year... @ethomson do you think you could give the issue a gentle poke internally? (Assuming you still work for Github!)

**Reply by craig-hill-SH** [reply]
Any updates that can be provided here? I'm running into the same issue.

**Comment by dinhnhat0401** [comment]
any updates on this?

**Comment by nhhai** [comment]
+1
Our use case is to add changes log / release notes into manual workflows.

**Reply by liviudobrea** [reply]
Same, would be nice to have

**Reply by M07amedRamadan** [reply]
same issue

**Comment by nhhai** [comment]
Currently, our workaround is to send `\n` as the newline character, and then use echo to paste it to file.
This is a step in the workflow:
```bash
    run: |
          rm -rf artifacts
          mkdir artifacts

          echo -e "$CHANGES_LOG" >> artifacts/changes_log.txt
          echo -e "$INTERNAL_NOTES" >> artifacts/internal_notes.txt

          echo "Changes log:"
          cat artifacts/changes_log.txt
          echo ""

          echo "Internal notes:"
          cat artifacts/internal_notes.txt
          echo ""
```
You can use your text editor to prepare the notes, then replace the newline character with `\n` to paste into the workflow.

**Comment by kr185198** [comment]
Hello @ethomson. We are having a use case that requires a multiline string as input. Any updates on this ?

**Comment by archon810** [comment]
Would love an update here as well. We're looking to add a changelog field on the fly for workflow runs and a single-line string is not good enough for that.

**Comment by jkester1986** [comment]
any update on this? It would be incredibly helpful

**Comment by dawilliams-gpsw** [comment]
+1 here as well. We have a similar need for multi line release notes/comments

**Comment by smesh** [comment]
+1. Example: sql script.

**Comment by Meenu-Chandrasekaran** [comment]
+1

**Comment by zevnda** [comment]
+1 the main reason I'm not using workflows for creating my releases is the lack of a multi-line textarea. It makes adding changelogs/release notes an annoyance. Sure, formatting a single line string of text is managable, but it is an unnecessary step, so I may as well manually create them myself

**Comment by nebuk89** [comment]
Sorry for Actions Product being quiet here, I will add this to our papercuts backlog that we are working against over the next 12 months. I cannot commit we will get to it :) but know that we are still aware (it has not been forgotten).

**Reply by dolma** [reply]
it has been exactly 1 year now, could you please provide an update as to whether this is still in the backlog, abandoned, or in progress?
Thank you

**Reply by grtcdr** [reply]
Yes please :)

**Reply by craig-hill-SH** [reply]
Any updates on this?

**Comment by AriPerkkio** [comment]
As work-around, you can change the element type with developer tools and paste multiline string just fine. It's usable in actions as is.

```diff
- <input
+ <textarea
   class="form-control
   input-contrast
   input-sm"
   type="text"
   id="input_0"
   required=""
   name="inputs[message]"
   value="">
```

<img src="https://github.com/user-attachments/assets/12ca67fc-0f1e-4c67-97d8-b750ce199bf9" width="400" />

**Reply by dawilliams-gpsw** [reply]
That's an excellent workaround @gabimoncha and also shows it should be pretty simple for Github to implement this feature. 🤞🏻

**Reply by jkester1986** [reply]
That would work....but that's not a great workaround IMO. Another alternative is to take what you want to have in the input, and put it into a string escape tool like https://appdevtools.com/json-escape-unescape. This is what my team is doing. But both of these workarounds are things that have to be done every time the workflow is run.

**Comment by arsovski-ahnl** [comment]
+1

**Comment by MOmarMiraj** [comment]
+1

**Comment by kindkitchen** [comment]
+100

**Comment by aducos** [comment]
+1

**Comment by dptsolutions** [comment]
+1. I too want to add release notes as an input to a workflow_dispatch trigger. I need multiline support.

**Comment by xdatsg** [comment]
+1

**Comment by tony-truong-wcq** [comment]
+1

**Comment by shubhamayrindigital** [comment]
+1

**Comment by kackogut** [comment]
+1

**Comment by tulio-o** [comment]
**Unbelievable** that a simple change from `input` to `textarea` make it work but the GitHub Actions teams is taking 2 long long years to add this simple feature.

**Reply by Timmmm** [reply]
3.5 years... so far.

**Comment by rumessh** [comment]
+1. with all these AI prompts and stuff, it will be convenient to input a whole prompt to the github workflow in text area while manually triggering an automated workflow.

**Comment by fire-at-will** [comment]
This would be a huge help! We'd like to use this to take in release notes to our workflows, but that's pretty tough to do without line breaks.

**Comment by zachicecreamcohn** [comment]
This would be incredibly helpful! The workaround of changing the `<input>` to a `<textarea>` in the dev tools works but is very very janky. Any updates on this?

**Comment by slavb18** [comment]
They can use copilot to fix this issue!

**Comment by shreyasbhs** [comment]
+1. `textarea` type would be a really useful option.

**Comment by NeilNagaraj** [comment]
+1. textarea type would be a really useful option.

**Comment by borrougagnou** [comment]
+1 Please !

@queenofcorgis any update for us maybe ?

**Reply by queenofcorgis** [reply]
Hey there! I know this discussion has been open for a while, we appreciate your patience. You can check our [public roadmap](https://github.com/orgs/github/projects/4247/views/1) to see the kinds of work we are focusing on for the coming year. Smaller fixes won’t be listed, but are often included when work is happening adjacent to it.

While it has been a while since we have updated on this thread, the absence of an update is simply that - there has been no update because there is so far nothing to report. That doesn't mean it isn't being worked on - it might simply be that we're unable to share details about work that isn't ready for release.

As a GitHub user, the most you can do is to share your feedback on this Discussion space with the correct tags (include examples), and trust that we are always working to make GitHub as awesome as possible. We can't guarantee that every bit of feedback sees action, but we are paying attention.

**Reply by Timmmm** [reply]
I think he meant a real update, not whatever this is.

**Comment by BacLuc** [comment]
I would also really like this feature.
+100
